Firebase/Messaging/Public/FIRMessaging.h

```diff
@@ -252,7 +252,8 @@ NS_SWIFT_NAME(MessagingDelegate)
 /// Handle data messages received via FCM direct channel (not via APNS).
 - (void)messaging:(FIRMessaging *)messaging
     didReceiveMessage:(FIRMessagingRemoteMessage *)remoteMessage
-NS_SWIFT_NAME(messaging(_:didReceive:));
+NS_SWIFT_NAME(messaging(_:didReceive:))
+__deprecated_msg("Downstream message in direct channel is deprecated, please use APNS channel.");
```

Comment:
Is there a URL that describes how to transition?

Reply:
Public doc probably launch at the same time, there's no direct alternative. Dev can use APNS for data message delivery and FireStore, firebase database, remote config for real time updates. But yeah we might be able to tuck in the public doc change for this.